### PR TITLE
Refine TFResource transform into DeployResource.

### LINKED ISSUE #243: Redefine TFResource transform into DeployResource.

### PR DESCRIPTION
**Pull request by baixinsui**
fix #243

### CODE REVIEW

**Comment by baixinsui** [comment]
> @baixinsui We are doing these changes for HuaweiCloud only primarily but we must also consider openstack equally because openstack and Huaweicloud are the only two main plugins that are in our current RC2 scope. Could you please see if we can add these changes for openstack too? It is ideally not too different to your Huaweicloud changes.

Sure. No problem. Can you give me a resource.tfstate file or a OCL2.0 resource.yml file of Opanstack service?

**Comment by baixinsui** [comment]
> this is one of the simple files. Please let me know if this is sufficient.
> 
> ` variable vm_name {} variable flavor_flavor_name {}
> 
> data "openstack_networking_network_v2" "network" { name = "public" }
> 
> resource "openstack_compute_instance_v2" "vm_ordered" { name = var.vm_name flavor_name = var.flavor_flavor_name image_name = "cirros-0.5.2-x86_64-disk"
> 
> network { uuid = data.openstack_networking_network_v2.network.network_id } } `

I hava got  more resource info from  https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs .

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/resource/TfStateResource.java [hunk 1]
```diff
@@ -21,5 +21,9 @@ public class TfStateResource {
 
     public String name;
 
+    public String mode;
+
+    private String provider;
```
Comment (iskey) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/resource/TfStateResource.java:26]: Has anyone used this field? What does this field mean to do?

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleTerraformResourceHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URL;
+import org.apache.commons.collections.CollectionUtils;
+import org.eclipse.xpanse.modules.deployment.deployers.terraform.resource.TfState;
+import org.eclipse.xpanse.modules.models.service.DeployResult;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class FlexibleTerraformResourceHandlerTest {
```
Comment (iskey) [inline plugins/flexibleengine/src/test/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleTerraformResourceHandlerTest.java:12]: Please use the fullname of `FlexibleEngine` here.

Comment (iskey) [inline plugins/flexibleengine/src/test/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleTerraformResourceHandlerTest.java:12]:
```suggestion
public class FlexibleEngineTerraformResourceHandlerTest {
```

Please use the fullname of `FlexibleEngine` here.

File: plugins/flexibleengine/src/test/resources/flexible-tfstate.json [hunk 3]
```diff
@@ -0,0 +1,712 @@
...
+            "volume_attached": [
+              {
+                "boot_index": 0,
+                "pci_address": "0000:02:01.0",
+                "size": 40,
+                "type": "SAS",
+                "uuid": "a0bf9552-c3f7-4f0a-a48f-796be76d4057"
+              }
+            ]
+          },
```
Comment (iskey) [inline plugins/flexibleengine/src/test/resources/flexible-tfstate.json:143]: Please remove the sensitive informations?

File: plugins/flexibleengine/src/test/resources/flexible-tfstate.json [hunk 4]
```diff
@@ -0,0 +1,712 @@
...
+    },
+    {
+      "mode": "managed",
+      "type": "flexibleengine_compute_keypair_v2",
+      "name": "keypair",
+      "provider": "provider[\"registry.terraform.io/flexibleenginecloud/flexibleengine\"]",
+      "instances": [
+        {
+          "schema_version": 0,
+          "attributes": {
```
Comment (iskey) [inline plugins/flexibleengine/src/test/resources/flexible-tfstate.json:442]: doitto, for the public_key.

File: plugins/openstack/src/test/resources/openstack-tfstate.json
Comment (iskey) [inline plugins/openstack/src/test/resources/openstack-tfstate.json:1]: Please remove the sensitive informations here.

**Review by iskey — CHANGES_REQUESTED** [review]
some comments added.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/resource/TfStateResource.java (hunk 1 above)
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/resource/TfStateResource.java]: I add note on field 'mode' and remove the unused field 'provider'

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleTerraformResourceHandlerTest.java (hunk 2 above)
Comment (baixinsui) [reply plugins/flexibleengine/src/test/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/FlexibleTerraformResourceHandlerTest.java:12]: Ok.

**Review by iskey — APPROVED** [review]
looks good but some sensive informations still need to be deleted.

File: plugins/huaweicloud/src/test/resources/huawei-tfstate.json [hunk 5]
```diff
@@ -0,0 +1,1103 @@
...
+            "id": "88cd9538-8db0-45da-8bee-ebd3916500df",
+            "name": "kafka-vpc-web-27e7c494",
+            "region": "cn-southwest-2",
+            "routes": [],
+            "secondary_cidr": null,
+            "status": "OK",
+            "tags": null,
+            "timeouts": null
+          },
+          "sensitive_attributes": [],
```
Comment (iskey) [inline plugins/huaweicloud/src/test/resources/huawei-tfstate.json:946]: please remove this private string.

**Review by iskey — CHANGES_REQUESTED** [review]
add some comments

**Pull request closed without merging**